apps/server/src/modules/h5p-editor/service/temporary-file-storage.service.ts

```diff
@@ -83,42 +114,43 @@ export class TemporaryFileStorage implements ITemporaryFileStorage {
 		expirationTime: Date
 	): Promise<ITemporaryFile> {
 		this.checkFilename(filename);
+
 		const now = new Date();
 		if (expirationTime < now) {
 			throw new NotAcceptableException('expirationTime must be in the future');
 		}
 
 		const path = this.getFilePath(user.id, filename);
-		let tempFile: H5pEditorTempFile | undefined;
+
 		try {
-			tempFile = await this.repo.findByUserAndFilename(user.id, filename);
 			await this.s3Client.delete([path]);
-		} finally {
-			if (tempFile === undefined) {
-				tempFile = new H5pEditorTempFile({
-					filename,
-					ownedByUserId: user.id,
-					expiresAt: expirationTime,
-					birthtime: new Date(),
-					size: dataStream.bytesRead,
-				});
-			} else {
-				tempFile.expiresAt = expirationTime;
-				tempFile.size = dataStream.bytesRead;
-			}
+		} catch (err) {
+			/* file did not exist, just continue */
```

Comment:
We really don't care about any errors here. This is a "just in case" remove, which should never happen because the filenames are randomly generated by H5P. Even if something went wrong here, the s3.create() call in line 136 would just override the file anyway

Reply:
Why remove you before creating, the "Create" function overwrites the object if file exists

Reply:
Removed the call to `delete`